web/packages/teleport/src/Discover/Server/DownloadScript/DownloadScript.tsx

```diff
@@ -137,6 +107,60 @@ export function DownloadScript(props: AgentStepProps) {
     props.nextStep();
   }
 
+  let hint;
+  if (showHint) {
+    hint = (
+      <HintBox>
+        <Text color="warning">
+          <Flex alignItems="center" mb={2}>
+            <TextIcon
+              color="warning"
+              css={`
+                white-space: pre;
+              `}
+            >
+              <Icons.Warning fontSize={4} color="warning" />
+            </TextIcon>
+            We're still looking for your server
+          </Flex>
+        </Text>
+
+        <Text mb={3}>
+          There are a couple of possible reasons for why we haven't been able to
+          detect your server.
+        </Text>
+
+        <Text mb={1}>
+          - The command was not run on the server you were trying to add.
+        </Text>
+
+        <Text mb={3}>
+          - The Teleport SSH Service could not join this Teleport cluster. Check
+          the logs for errors by running <Mark>journalctl status teleport</Mark>
+          .
+        </Text>
+
+        <Text>
+          We'll continue to look for the server whilst you diagnose the issue.
+        </Text>
+      </HintBox>
+    );
+  } else {
+    hint = (
+      <WaitingInfo>
+        <TextIcon
+          css={`
+            white-space: pre;
+          `}
+        >
+          <Icons.Restore fontSize={4} />
+        </TextIcon>
+        After running the command above, we'll automatically detect your new
+        Teleport instance.
+      </WaitingInfo>
+    );
+  }
```

Comment:
Something similar pops up multiple times, it looks like something we can parameterize with props (including the timeout) and hold in `web/packages/teleport/src/Discover/Shared`.

Reply:
I've simplified the usage of `HintBox` and moved the common hook behaviour into a new `useShowHint()`